File: Practice/AOJ/ITP1/ITP1_6C/itp1_6c.py
billdings = [[[0 for _ in range(10)] for _ in range(3)] for _ in range(4)]


def printstate(billdings):
    for i in range(len(billdings)):
        for j in range(len(billdings[0])):
            for k in range(len(billdings[0][0])):
                print(" "+str(billdings[i][j][k]), end="")
            print()
        if i != len(billdings)-1:
            print("#"*20)


n = int(input())
for _ in range(n):
    b, f, r, v = map(int, input().split())
    b, f, r = b-1, f-1, r-1
    billdings[b][f][r] += v

printstate(billdings)

File: Practice/AOJ/ITP1/ITP1_6C/itp1_6c.rb
billdings = Array.new(4).map{Array.new(3).map{Array.new(10, 0)}}
def printstate(b)
    b.each.with_index(1) do |bill, bill_idx|
        bill.each do |floor|
            floor.each do |room|
                print " " + room.to_s
            end
            puts
        end
        if bill_idx != b.size then puts "#"*20 end
    end
end

n = gets.to_i
n.times do
    b, f, r, v = gets.chomp.split.map(&:to_i)
    billdings[b-1][f-1][r-1]+=v
end

printstate(billdings)